starknet_py/transaction_exceptions.py

```diff
@@ -1,13 +1,16 @@
+from typing import Optional, cast
+
+
 class TransactionFailedError(Exception):
     """
     Base exception for transaction failure
     """
 
     def __init__(
         self,
-        message: str = "Unknown starknet error",
+        message: Optional[str] = None,
     ):
-        self.message = message
+        self.message = cast(str, message or "Unknown starknet error")
```

Comment:
```suggestion
        if message is None:
            message = "Unknown starknet error"
        self.message = message
```

Reply:
Why? If this is an empty string then it might as well be turned into "Unknown starknet error"

Reply:
To make pyright happy without having to do typecasting. I didn't consider the fact that we'd also like to replace `""` with default message.